docs/source/en/training.mdx

```diff
@@ -151,66 +151,113 @@ Then fine-tune your model by calling [`~transformers.Trainer.train`]:
+This approach is great when it works, but for larger datasets you might find it starting to become a problem. Why?
+Because the tokenized array and labels would have to be fully loaded into memory, and because Numpy doesn’t handle
+“jagged” arrays, so every tokenized sample would have to be padded to the length of the longest sample in the whole
+dataset. That’s going to make your array even bigger, and all those padding tokens will slow down training too!
+
+### Loading data as a tf.data.Dataset
+
+If you want to avoid this, you can load your data as a `tf.data.Dataset` instead. Although you can write your own
+`tf.data` pipeline if you want, we have two convenience methods for doing this - [`~datasets.Dataset.to_tf_dataset`]
+and [`~TFPreTrainedModel.prepare_tf_dataset`]. We recommend `prepare_tf_dataset()` in most cases because it is simpler
+and can automatically infer which columns from the dataset to load by comparing them against the input names for the
+model. The `to_tf_dataset()` method is more low-level, and is useful when you want to exactly control how your
+dataset is created.
+
+Before you can use this method, you will need to add the tokenizer outputs to your dataset as columns, as in
+the following code sample:
 
 ```py
->>> tf_train_dataset = small_train_dataset.to_tf_dataset(
-...     columns=["attention_mask", "input_ids", "token_type_ids"],
-...     label_cols=["labels"],
-...     shuffle=True,
-...     collate_fn=data_collator,
-...     batch_size=8,
-... )
+def tokenize_dataset(data):
+    # Keys of the returned dictionary will be added to the dataset as columns
+    return tokenizer(data["text"])
 
->>> tf_validation_dataset = small_eval_dataset.to_tf_dataset(
-...     columns=["attention_mask", "input_ids", "token_type_ids"],
-...     label_cols=["labels"],
-...     shuffle=False,
-...     collate_fn=data_collator,
-...     batch_size=8,
-... )
+
+dataset = dataset.map(tokenize_dataset)
 ```
 
-### Compile and fit
+Remember that Hugging Face datasets are stored on disk by default, so this will not inflate your memory usage! Once the
+columns have been added, we can now stream batches from the dataset and add padding on the fly, which greatly
+reduces the number of padding tokens compared to padding the entire dataset.
 
-Load a TensorFlow model with the expected number of labels:
 
 ```py
->>> import tensorflow as tf
->>> from transformers import TFAutoModelForSequenceClassification
-
->>> model = TFAutoModelForSequenceClassification.from_pretrained("bert-base-cased", num_labels=5)
+>>> tf_dataset = model.prepare_tf_dataset(dataset, batch_size=16, shuffle=True, tokenizer=tokenizer)
 ```
 
-Then compile and fine-tune your model with [`fit`](https://keras.io/api/models/model_training_apis/) as you would with any other Keras model:
+<Tip>
+
+The reason we pass the tokenizer to `prepare_tf_dataset` is so that it can correctly pad batches as they're loaded.
```

Comment:
This is a pretty substantial paragraph, so I think it's better to include it in the main body of content instead of as a Tip.

```suggestion
You need to pass the tokenizer to `prepare_tf_dataset` so it can correctly pad batches as they're loaded.
```

Reply:
Done! I also included those edits.